libcxx.mixin

```diff
@@ -0,0 +1,10 @@
+build:
+  libcxx:
+    cmake-args:
+      - "-DCMAKE_C_COMPILER=clang-6.0"
+      - "-DCMAKE_CXX_COMPILER=clang++-6.0"
+      - "-DCMAKE_CXX_FLAGS=-stdlib=libc++ -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS"
```

Comment:
Is there any reason someone might want to use libcxx without the thread safety annotations enabled? That is to say, is the enablement of that something we should consider splitting out? (I've not considered at all whether mixins are easy to blend together)

Reply:
The definition only adds annotations to the stdlib headers - it doesn't enable analysis. The way I understand it, this is a no-op when using clang, when `-Wthread-safety` is not specified. If we set it up this way, upstream packages only need to add that `-W` flag to get the analysis. Having a single flag to enable upstream seems like the best experience to me.